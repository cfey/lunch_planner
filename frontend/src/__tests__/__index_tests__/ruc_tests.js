import React from 'react'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import 'isomorphic-fetch'
import Ruc from '../../pages/index/components/ruc'

describe('Ruc', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MuiThemeProvider><Ruc /></MuiThemeProvider>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  /*
  // TODO: Mock to get user number
  // TODO: Mount ruc component
  it('counts correctly', async () => {
    const ruc = new Ruc()
    const count = await ruc.getRegisteredUserNumber()
    expect(count).toEqual(6)
  })
  */
})
