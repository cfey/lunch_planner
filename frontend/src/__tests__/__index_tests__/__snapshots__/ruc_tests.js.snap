// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ruc renders correctly 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          Registered User Counter
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      Registered User: 
      <b>
        -
      </b>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#00bcd4",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            refresh
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
